llvm/lib/SYCLLowerIR/LowerESIMD.cpp

```diff
@@ -1210,10 +1210,11 @@ PreservedAnalyses SYCLLowerESIMDPass::run(Function &F,
     if (auto CastOp = dyn_cast<llvm::CastInst>(&I)) {
       llvm::Type *DstTy = CastOp->getDestTy();
       auto CastOpcode = CastOp->getOpcode();
-      if ((CastOpcode == llvm::Instruction::FPToUI &&
-           DstTy->getScalarType()->getPrimitiveSizeInBits() <= 32) ||
+      if (isa<FixedVectorType>(DstTy) &&
```

Comment:
Why not just check for isVectorTy()? Does front-end generate ScalableVectorType?

Reply:
> Why not just check for isVectorTy()?

VectorType llvm type has been split into VectorType and FixedVectorType type for known vector widths throughout LLVM some time ago. gpu::simd is always a FixedVectorType.

> Does front-end generate ScalableVectorType?

I'm not sure. How is this relevant - can you please elaborate?

Reply:
ok, so it's always FixedVectorType. I was just wondering if ScalableVectorType is supported.